This does not successfully publish here:

ginstall -c -m 644 Source/FreeImage.h /export/home/andreas/oi-userland/components/library/FreeImage/build/prototype/i386/usr/include
ginstall: cannot stat '/export/home/andreas/oi-userland/components/library/FreeImage/build/prototype/i386/usr/include/FreeImage.h': Permission denied
make[2]: *** [Makefile.gnu:75: install] Error 1

As a side note: Makefile, FreeImage.p5m, and libfreeimage.pc have executable permissions.

--- a/components/library/FreeImage/FreeImage.p5m
+++ b/components/library/FreeImage/FreeImage.p5m
@@ -1,0 +1,29 @@
+#
+# This file and its contents are supplied under the terms of the
+# Common Development and Distribution License ("CDDL"), version 1.0.
+# You may only use this file in accordance with the terms of version
+# 1.0 of the CDDL.
+#
+# A full copy of the text of the CDDL should have accompanied this
+# source.  A copy of the CDDL is also available via the Internet at
+# http://www.illumos.org/license/CDDL.
+#
+
+#
+# Copyright 2022, Daniel Chan
+#
+
+set name=pkg.fmri value=pkg:/$(COMPONENT_FMRI)@$(IPS_COMPONENT_VERSION),$(BUILD_VERSION)
+set name=pkg.summary value="$(COMPONENT_SUMMARY)"
+set name=info.classification value="$(COMPONENT_CLASSIFICATION)"
+set name=info.upstream-url value=$(COMPONENT_PROJECT_URL)
+set name=info.source-url value=$(COMPONENT_ARCHIVE_URL)
+set name=org.opensolaris.consolidation value=$(CONSOLIDATION)
+
+license $(COMPONENT_LICENSE_FILE) license='$(COMPONENT_LICENSE)'
+
+file path=usr/include/FreeImage.h
+file path=usr/lib/$(MACH64)/libfreeimage-3.18.0.so
+link path=usr/lib/$(MACH64)/libfreeimage.so target=libfreeimage.so.3
+link path=usr/lib/$(MACH64)/libfreeimage.so.3 target=libfreeimage-3.18.0.so
+file path=usr/lib/$(MACH64)/pkgconfig/libfreeimage.pc

--- a/components/library/FreeImage/Makefile
+++ b/components/library/FreeImage/Makefile
@@ -1,0 +1,51 @@
+#
+# This file and its contents are supplied under the terms of the
+# Common Development and Distribution License ("CDDL"). You may
+# only use this file in accordance with the terms of the CDDL.
+#
+# A full copy of the text of the CDDL should have accompanied this
+# source. A copy of the CDDL is also available via the Internet at
+# http://www.illumos.org/license/CDDL.
+#
+
+#
+# Copyright 2022, Daniel Chan
+#
+
+BUILD_BITS=		64
+BUILD_STYLE=	justmake
+
+include ../../../make-rules/shared-macros.mk
+
+COMPONENT_NAME=		FreeImage
+COMPONENT_VERSION_MAJOR=3
+COMPONENT_VERSION_MINOR=18
+COMPONENT_VERSION_PATCH=0
+COMPONENT_VERSION=	$(COMPONENT_VERSION_MAJOR).$(COMPONENT_VERSION_MINOR).$(COMPONENT_VERSION_PATCH)
+COMPONENT_SUMMARY=	A library project for developers to support popular graphics image formats
+COMPONENT_PROJECT_URL=	https://freeimage.sourceforge.io
+COMPONENT_FMRI=		library/libfreeimage
+COMPONENT_CLASSIFICATION=System/Libraries
+COMPONENT_SRC=		$(COMPONENT_NAME)
+COMPONENT_ARCHIVE=	$(COMPONENT_NAME)$(COMPONENT_VERSION_MAJOR)$(COMPONENT_VERSION_MINOR)$(COMPONENT_VERSION_PATCH).zip
+COMPONENT_ARCHIVE_URL=	http://downloads.sourceforge.net/freeimage/$(COMPONENT_ARCHIVE)
+COMPONENT_ARCHIVE_HASH=	sha256:f41379682f9ada94ea7b34fe86bf9ee00935a3147be41b6569c9605a53e438fd
+COMPONENT_LICENSE=	GPLv3
+COMPONENT_LICENSE_FILE=	license-gplv3.txt
+
+include $(WS_MAKE_RULES)/common.mk
+
+COMPONENT_POST_UNPACK_ACTION=cp $(SOURCE_DIR)/../libfreeimage.pc $(SOURCE_DIR)/
+COMPONENT_BUILD_GMAKE_ARGS="file=Makefile.gnu"
+CFLAGS+=-fPIC
+CFLAGS+=-fexceptions
+CFLAGS+=-fvisibility=hidden
+CXXFLAGS+=-fPIC 
+CXXFLAGS+=-fexceptions
+CXXFLAGS+=-fvisibility=hidden 
+CXXFLAGS+=-Wno-ctor-dtor-privacy
+
+# Auto-generated dependencies
+REQUIRED_PACKAGES += $(GCC_RUNTIME_PKG)
+REQUIRED_PACKAGES += $(GXX_RUNTIME_PKG)
+REQUIRED_PACKAGES += system/library

--- a/components/library/FreeImage/libfreeimage.pc
+++ b/components/library/FreeImage/libfreeimage.pc
@@ -1,0 +1,11 @@
+prefix=/usr
+exec_prefix=${prefix}
+libdir=/usr/lib/amd64
+includedir=${prefix}/include
+Libs_Static=/usr/lib/amd64/libfreeimage.a
+
+Name: libfreeimage
+Version: 21.9
+Description: FreeImage
+Libs: -L${libdir} -lfreeimage
+Cflags: -I${includedir}

--- a/components/library/FreeImage/manifests/sample-manifest.p5m
+++ b/components/library/FreeImage/manifests/sample-manifest.p5m
@@ -1,0 +1,30 @@
+#
+# This file and its contents are supplied under the terms of the
+# Common Development and Distribution License ("CDDL"), version 1.0.
+# You may only use this file in accordance with the terms of version
+# 1.0 of the CDDL.
+#
+# A full copy of the text of the CDDL should have accompanied this
+# source.  A copy of the CDDL is also available via the Internet at
+# http://www.illumos.org/license/CDDL.
+#
+
+#
+# Copyright 2022 <contributor>
+#
+
+set name=pkg.fmri value=pkg:/$(COMPONENT_FMRI)@$(IPS_COMPONENT_VERSION),$(BUILD_VERSION)
+set name=pkg.summary value="$(COMPONENT_SUMMARY)"
+set name=info.classification value="$(COMPONENT_CLASSIFICATION)"
+set name=info.upstream-url value=$(COMPONENT_PROJECT_URL)
+set name=info.source-url value=$(COMPONENT_ARCHIVE_URL)
+set name=org.opensolaris.consolidation value=$(CONSOLIDATION)
+
+license $(COMPONENT_LICENSE_FILE) license='$(COMPONENT_LICENSE)'
+
+file path=usr/include/FreeImage.h
+file path=usr/lib/$(MACH64)/libfreeimage-$(COMPONENT_VERSION).so
+link path=usr/lib/$(MACH64)/libfreeimage.so target=libfreeimage.so.3
+link path=usr/lib/$(MACH64)/libfreeimage.so.3 \
+    target=libfreeimage-$(COMPONENT_VERSION).so
+file path=usr/lib/$(MACH64)/pkgconfig/libfreeimage.pc

--- a/components/library/FreeImage/patches/01-Makefile.gnu.patch
+++ b/components/library/FreeImage/patches/01-Makefile.gnu.patch
@@ -1,0 +1,28 @@
+--- FreeImage/Makefile.gnu	Tue Mar 10 09:04:00 2015
++++ FreeImage/Makefile.gnu	Sat Feb 26 00:22:13 2022
+@@ -6,7 +6,8 @@
+ # General configuration variables:
+ DESTDIR ?= /
+ INCDIR ?= $(DESTDIR)/usr/include
+-INSTALLDIR ?= $(DESTDIR)/usr/lib
++INSTALLDIR ?= $(DESTDIR)/usr/lib/amd64
++PKGCFGDIR ?= $(DESTDIR)/usr/lib/amd64/pkgconfig
+ 
+ # Converts cr/lf to just lf
+ DOS2UNIX = dos2unix
+@@ -70,10 +71,11 @@
+ 	$(CC) -s -shared -Wl,-soname,$(VERLIBNAME) $(LDFLAGS) -o $@ $(MODULES) $(LIBRARIES)
+ 
+ install:
+-	install -d $(INCDIR) $(INSTALLDIR)
+-	install -m 644 -o root -g root $(HEADER) $(INCDIR)
+-	install -m 644 -o root -g root $(STATICLIB) $(INSTALLDIR)
+-	install -m 755 -o root -g root $(SHAREDLIB) $(INSTALLDIR)
++	ginstall -c -d $(INCDIR) $(PKGCFGDIR)
++	ginstall -c -m 644 $(HEADER) $(INCDIR)
++	
++	ginstall -c -m 755 $(SHAREDLIB) $(INSTALLDIR)
++	ginstall -c -m 644 libfreeimage.pc $(PKGCFGDIR)
+ 	ln -sf $(SHAREDLIB) $(INSTALLDIR)/$(VERLIBNAME)
+ 	ln -sf $(VERLIBNAME) $(INSTALLDIR)/$(LIBNAME)	
+ #	ldconfig

--- a/components/library/FreeImage/pkg5
+++ b/components/library/FreeImage/pkg5
@@ -1,0 +1,13 @@
+{
+    "dependencies": [
+        "SUNWcs",
+        "shell/ksh93",
+        "system/library",
+        "system/library/g++-7-runtime",
+        "system/library/gcc-7-runtime"
+    ],
+    "fmris": [
+        "library/libfreeimage"
+    ],
+    "name": "FreeImage"
+}

